polygon/p2p/fetcher.go

```diff
@@ -2,18 +2,24 @@ package p2p
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"time"
 
 	"github.com/ledgerwatch/log/v3"
+	"modernc.org/mathutil"
 
+	"github.com/ledgerwatch/erigon-lib/common"
 	"github.com/ledgerwatch/erigon-lib/gointerfaces/sentry"
 	"github.com/ledgerwatch/erigon/core/types"
 	"github.com/ledgerwatch/erigon/eth/protocols/eth"
 	"github.com/ledgerwatch/erigon/rlp"
 )
 
-const responseTimeout = 5 * time.Second
+const (
+	responseTimeout      = 5 * time.Second
+	maxFetchHeadersRange = 16384
```

Comment:
what's the reason for this limit? I feel that it should be up to the client to decide

Reply:
this was so that we have an upper bound on number of requests sent to a peer at the same time - it is no longer applicable since we now changed to do 1 request at a time when chunking so ive removed it